graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -44,6 +51,138 @@ def _are_filter_fields_uniquely_indexed(filter_fields, unique_indexes):
+
+    return parameters[parameter_name]
+
+
+def _convert_uuid_string_to_int(uuid_string):
+    """Return the integer representation of a UUID string."""
+    return UUID(uuid_string).int
+
+
+def _get_selectivity_of_integer_inequality_filter(
+    domain_interval, parameter_values, filter_operator
+):
+    """Return the selectivity of a given integer inequality filter filtering over a given interval.
+
+    Args:
+        domain_interval: tuple of (int, int), describing the inclusive lower and upper bound of the
+                         domain of integers being filtered.
+        parameter_values: tuple of integers, describing the parameters for the inequality filter.
+        filter_operator: str, describing the inequality filter operation being performed.
+
+    Returns:
+        Selectivity object, describing the selectivity of the integer inequality filter.
+    """
+    # Inequality filters on integers can also be represented by an interval for the subset of values
+    # that pass through the filter. By representing the filter as an interval, we avoid considering
+    # the operators separately.
+    query_interval = (domain_interval[0], domain_interval[1])
+    if filter_operator == '>':
+        # This (exclusively) constrains the lower bound of the values passing through the filter.
+        query_interval[0] = max(query_interval[0], parameter_values[0] + 1)
+    elif filter_operator == '>=':
+        # This (inclusively) constrains the lower bound of the values passing through the filter.
+        query_interval[0] = max(query_interval[0], parameter_values[0])
+    elif filter_operator == '<':
+        # This (exclusively) constrains the upper bound of the values passing through the filter.
+        query_interval[1] = min(query_interval[1], parameter_values[0] - 1)
+    elif filter_operator == '<=':
+        # This (inclusively) constrains the upper bound of the values passing through the filter.
+        query_interval[1] = min(query_interval[1], parameter_values[0])
+    elif filter_operator == 'between':
+        # This (inclusively) constrains the lower and upper bounds of the
+        # values passing through the filter.
+        query_interval[0] = max(query_interval[0], parameter_values[0])
+        query_interval[1] = min(query_interval[1], parameter_values[1])
+    else:
+        raise AssertionError(u'Cost estimator found unsupported integer inequality operator {}'
+                             .format(filter_operator))
+
+    # If the interval of values passing through the filters is empty, no results will be
+    # returned. This happens if the interval's upper bound is smaller than the lower bound.
+    if query_interval[0] > query_interval[1]:
+        field_selectivity = Selectivity(kind=ABSOLUTE_SELECTIVITY, value=0.0)
+        return field_selectivity
+
+    if not domain_interval[0] <= query_interval[0] <= query_interval[1] <= domain_interval[1]:
```

Comment:
This is provably impossible to trigger, but I want to leave it in in case this code gets changed later. Thoughts?